### ISSUE TITLE
[Feature Request] Broken out section views

### ISSUE THREAD

**Issue by FEA-eng** [FreeCAD/FreeCAD#8075]
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?f=35&t=74105

### Subproject(s) affected?

Techdraw

### Idea description

The new Complex Section tool added support for several advanced section types. However, broken out sections are not yet supported and have to be created manually with workarounds.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by MakerFC** [comment]
please also add Forums discussion:
[Techdraw broken-out section]
(https://forum.freecadweb.org/viewtopic.php?f=35&t=74441)

**Comment by luzpaz** [comment]
Added to OP

**Comment by NXTler** [comment]
It's used a lot in the professional space, we should implement it.

**Comment by luzpaz** [comment]
Is this actually a duplicate of #5694 ?

**Comment by FEA-eng** [comment]
> Is this actually a duplicate of #5694 ?

No, despite similar naming, those are different kinds of views. The one discussed here is basically about looking locally inside the structure while the other (older) one is about depicting long parts by cutting a segment from the middle and putting the pieces close to each other.

**Comment by FEA-eng** [comment]
@pierreporte In SW, you can either specify the depth value or select the geometry in a drawing view up to which the section will be made.

![geom](https://github.com/FreeCAD/FreeCAD/assets/59876896/b4cd4a57-5b74-44d9-bb2f-d4a42b16c968)

I wouldn't mind if it was only possible to define the depth value in FreeCAD initially but geometry selection would be nice too. Fortunately, this feature should be possible to implement in a similar way as other advanced sections.

**Comment by pure-machinery** [comment]
Perhaps the creation should stay consistent with the other methods of defining cross sections. The distance is a no-no from my side.

On the other hand this has more to do with rendering in my opinion then the cross section code, since the cross section code is already here the only problem is how to display the local area of a section cut while all else is preserved. 

Multiple local cross sections are also possible and useful.

**Comment by 3nt3** [comment]
If the PR about this was merged, why is the issue sti open ^^

**Comment by FEA-eng** [comment]
> If the PR about this was merged, why is the issue still open ^^

It’s explained in the description of this issue - the merged PR implemented other types of advanced section views, not broken out ones discussed here.

**Comment by pierreporte** [comment]
A **150 € reward** will be given to the contributor that writes a pull request, solving this issue, that gets merged. It will be paid by the FreeCAD Project Association. It is to be considered an incentive. Grant applications are welcome, but the total amount will exclude the reward.

**Comment by Rajesh270712** [comment]
Hi, I opened a focused PR for this: https://github.com/FreeCAD/FreeCAD/pull/30863.

It adds an explicit-depth broken-out complex-section strategy for closed profiles, with a focused TechDraw regression test. Validation covered the TechDraw build targets, `TDTest.DrawComplexSectionTest.DrawComplexSectionTest`, `TestTechDrawGui`, and a semantic broken-out depth probe. If this approach looks aligned, could you assign the issue to @Rajesh270712? Happy to adjust if you prefer a different direction.

**Comment by Rajesh270712** [comment]
Hi, I opened the replacement PR here: https://github.com/FreeCAD/FreeCAD/pull/30976.

It keeps the focused TechDraw BrokenOut complex-section change on the replacement branch; the PR thread now includes a screenshot/setup note for the broken-out behavior. Validation covered `TDTest.DrawComplexSectionTest.DrawComplexSectionTest` plus the focused visual export check (`Up-to-date` section with visible edges). If this approach looks aligned, could you assign this issue to @Rajesh270712? Happy to adjust if you prefer a different direction.

**Comment by maxwxyz** [comment]
@PaddleStroke @FEA-eng I think this could work if we have the sketch in TD. Maybe some face or closed loop can be selected for a breakout and the depth can be defined?

**Comment by FEA-eng** [comment]
> [@PaddleStroke](https://github.com/PaddleStroke) [@FEA-eng](https://github.com/FEA-eng) I think this could work if we have the sketch in TD. Maybe some face or closed loop can be selected for a breakout and the depth can be defined?

This is how broken views and complex section views already work (you draw the cut profile and select it along with the view in TD), so it should be doable. It would only need the depth setting.

**Comment by PaddleStroke** [comment]
I'll check this out as well. 
If you guys have videos of another cad which you like the UI, I can replicate it.

**Comment by FEA-eng** [comment]
> I'll check this out as well. If you guys have videos of another cad which you like the UI, I can replicate it.

The key aspect here is that it’s a section in the form of a small cutout within the view. It’s like looking inside the model through this local cutout without creating a separate section view.

Here’s SW documentation: https://help.solidworks.com/2026/english/solidworks/sldworks/c_broken_out_section.htm

And a video: https://youtu.be/pu5GK39ahB0?is=X3gfuileZ_owv_Ea

This one and rotated/removed section views are IMO the last missing fundamental drawing features in TD. But this one here is much more common (and thus more important), from what I know.

**Comment by FEA-eng** [comment]
Here’s Mangojelly’s tedious workaround for this: https://youtu.be/Dg2szQ3xpAw?t=1104&is=PqQO8R23H8b0dIh3

**Comment by PaddleStroke** [comment]
Ok this is clear to me. Let me summarize and please confirm I got it right.

This broken out section tool should : 
- modify the current view. Not create a new view. (same as the break in my rework)
- When the tool launch the user draw a bspline (or a free hand drawing that we turn as a bspline? Maybe a bspline is cleaner)
- A spinbox to enter the depth of the cut.

Basically like the solidworks tool? Or what?

**Comment by FEA-eng** [comment]
> * modify the current view. Not create a new view. (same as the break in my rework)

Yes, here's the translated definition from a classic Polish textbook:

_Broken out sections are drawn as local sections on the views of the objects and are bounded by wavy or zigzag lines._

> * When the tool launch the user draw a bspline (or a free hand drawing that we turn as a bspline? Maybe a bspline is cleaner)

Wouldn't it be easiest if the tool used a sketch (with a closed profile) created before activating it? Lik what is already done for Complex Section and Broken Out view.

> * A spinbox to enter the depth of the cut.

Yes. See [here](https://github.com/FreeCAD/FreeCAD/issues/8075#issuecomment-2048163936). Some software also accepts geometrical features (e.g. datum planes) as references, but depth setting would still be needed.

**Comment by PaddleStroke** [comment]
> Wouldn't it be easiest if the tool used a sketch (with a closed profile) created before activating it? Lik what is already done for Complex Section and Broken Out view.

It's also possible, but much less user friendly I believe.

> Yes. See https://github.com/FreeCAD/FreeCAD/issues/8075#issuecomment-2048163936. Some software also accepts geometrical features (e.g. datum planes) as references, but depth setting would still be needed.

Ok let's start with a depth first and see about selecting geometrical feature after.

**Comment by FEA-eng** [comment]
> It's also possible, but much less user friendly I believe.

Yes, only easier to implement (especially since such approach already exists in TD).

> Ok let's start with a depth first and see about selecting geometrical feature after.

Sure, that will be perfectly sufficient for now.

**Comment by PaddleStroke** [comment]
<img width="1247" height="1085" alt="Image" src="https://github.com/user-attachments/assets/01041a0e-0c6c-48bf-9ab9-38beedff4798" />
Making progress :)

**Comment by PaddleStroke** [comment]
Working !

Video does not upload unfortunately.
I'll post a PR in 2 weeks when I'm back to home. There is already so many PR to review anyway.

**Comment by FEA-eng** [comment]
> Working !
> 
> Video does not upload unfortunately. I'll post a PR in 2 weeks when I'm back to home. There is already so many PR to review anyway.

Great! Please ping me there, and I'll prioritize it.
